### PR TITLE
Fix slashes get converted to underscores in the metrics UI page

### PR DESCRIPTION
**Pull request by jiacheliu3**
Removed escape() logic in alluxio.metrics.MetricsSystem of replacing "/" with "_".
Now the escape only replaces "." with "_".

### CODE REVIEW

**Comment by jiacheliu3** [comment]
Existing tests passed but I will add more tests corresponding to the added feature. Please don't merge this yet.

File: core/common/src/test/java/alluxio/metrics/MetricsSystemTest.java [hunk 1]
```diff
@@ -54,4 +57,61 @@ public void metricsSystem() {
...
+  /**
+   * Tests the "/" "." and "%" special characters can be escaped.
+   */
+  @Test
+  public void testEscape() {
+    AlluxioURI localUri1 = new AlluxioURI("/foo/alluxio/underFSStorage");
+    String localUriEscaped1 = MetricsSystem.escape(localUri1);
+    assertEquals("%2Ffoo%2Falluxio%2FunderFSStorage",
+            localUriEscaped1);
+    assertFalse(localUriEscaped1.contains("/"));
```
Comment (bf8086) [inline core/common/src/test/java/alluxio/metrics/MetricsSystemTest.java]: This is redundant - the `assertEquals` already checked that `localUriEscaped1` does not have `/`.

Same goes to the other `assertFalse` checks

Comment (jiacheliu3) [reply core/common/src/test/java/alluxio/metrics/MetricsSystemTest.java]: Redundant assertFalse() removed

File: core/common/src/test/java/alluxio/metrics/MetricsSystemTest.java [hunk 2]
```diff
@@ -25,7 +28,7 @@
 public final class MetricsSystemTest {
   private MetricsConfig mMetricsConfig;
   private static Counter sCounter =
-      MetricsSystem.METRIC_REGISTRY.counter(MetricsSystem.getMetricName("counter"));
+          MetricsSystem.METRIC_REGISTRY.counter(MetricsSystem.getMetricName("counter"));
```
[the remark this answers is not in the document]
Comment (jiacheliu3) [reply core/common/src/test/java/alluxio/metrics/MetricsSystemTest.java]: removed

File: core/server/master/src/main/java/alluxio/master/meta/AlluxioMasterRestServiceHandler.java [hunk 3]
```diff
@@ -997,9 +997,10 @@ public Response getWebUIMetrics() {
         }
         String ufs = metric.getTags().get(WorkerMetrics.TAG_UFS);
         if (isMounted(ufs)) {
+          // Unescape the URI for display
           Map<String, Long> perUfsMap = ufsOpsMap.getOrDefault(ufs, new TreeMap<>());
-          perUfsMap.put(ufs, (long) metric.getValue());
-          ufsOpsMap.put(ufs, perUfsMap);
+          perUfsMap.put(MetricsSystem.unescape(ufs), (long) metric.getValue());
+          ufsOpsMap.put(MetricsSystem.unescape(ufs), perUfsMap);
```
[the remark this answers is not in the document]
Comment (jiacheliu3) [reply core/server/master/src/main/java/alluxio/master/meta/AlluxioMasterRestServiceHandler.java]: reused as a local variable

File: core/common/src/main/java/alluxio/metrics/MetricsSystem.java [hunk 4]
```diff
@@ -357,14 +357,26 @@ public static String stripInstanceAndHost(String metricsName) {
   }
 
   /**
-   * Escapes a URI, replacing "/" and "." with "_" so that when the URI is used in a metric name,
-   * the "/" and "." won't be interpreted as path separators.
+   * Escapes a URI, replacing "." with "%2E" and "/" with "%2F".
+   * Replaces "%" with "%25" as well.
+   * So when the URI is used in a metric name, the "." and "/" won't be interpreted as
+   * path separators unescaped and interfere with the internal logic of AlluxioURI.
```
[the remark this answers is not in the document]
Comment (jiacheliu3) [reply core/common/src/main/java/alluxio/metrics/MetricsSystem.java]: comment improved slightly

**Comment by jiacheliu3** [comment]
Jenkins, test this please

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Comment by calvinjia** [comment]
alluxio-bot, cherry-pick this to branch-2.0 please.

**Pull request merged**